### PR TITLE
Vlad - wrote a set of unittests for ArrayUtils

### PR DESCRIPTION
**Pull request by vpatryshev**
Fixed ArrayUtils (remove() was broken)
Added more remove() versions, for a bunch of primitives
Changed one of Frame.remove()s so it uses ArrayUtils.remove() (others to follow)
Addedd a test for this Frame.remove()
Blocked a couple of NPEs in SVMLightFVecParseWriter (they were happening)
in Frame.get() added a better exception message for the case index is out of bounds (happens in a sparse example, will have to investigate it carefully)
Fixed a bunch of smart intellij warnings.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/272)

### CODE REVIEW

**Comment by vpatryshev** [comment]
This is an updated PR. The previous one was from a bad branch.

**Comment by arnocandel** [comment]
Did your private jenkins tests pass?

File: h2o-core/src/main/java/water/util/ArrayUtils.java [hunk 1]
```diff
@@ -1278,7 +1299,10 @@ public static Frame frame(Key key, String[] names, double[]... rows) {
     return fr;
   }
   public static Frame frame(double[]... rows) { return frame(null, rows); }
-  public static Frame frame(String[] names, double[]... rows) { return frame(Key.make(), names, rows); }
+  public static Frame frame(String[] names, double[]... rows) {
+    Key<Frame> k = Key.make();
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/util/ArrayUtils.java]: Minor comment: Just curious  about motivation here to change the code?

**Comment by vpatryshev** [comment]
It says "all checks have passed".

Thanks,
-Vlad

File: h2o-core/src/test/java/water/TestUtil.java [hunk 2]
```diff
@@ -376,6 +375,8 @@ public static Vec vec(String[] domain, int ...rows) {
     for (int i=0; i<a.length;i++) r[i][0] = a[i];
     return r;
   }
+
+  @SafeVarargs
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/TestUtil.java]: Is this Java6 safe?

**Comment by vpatryshev** [comment]
Good question. We are targeting Java6? Good to know. Commented out.
How come jenkins does not notice it...

Thanks,
-Vlad

File: h2o-core/src/main/java/water/util/ArrayUtils.java (hunk 1 above)
Comment (vpatryshev) [inline h2o-core/src/main/java/water/util/ArrayUtils.java]: "Type safety". Intellij was complaining about typeless instances of Key.

File: h2o-core/src/test/java/water/TestUtil.java (hunk 2 above)
Comment (vpatryshev) [inline h2o-core/src/test/java/water/TestUtil.java]: Commented out. Was not aware we target Java6.

**Comment by vpatryshev** [comment]
Regarding "writing all tests" - only when I make changes and the code has
no tests. Let others have the opportunity too.

Thanks,
-Vlad

File: h2o-core/src/main/java/water/util/ArrayUtils.java [hunk 3]
```diff
@@ -1333,9 +1357,31 @@ public static int countNonzeros(double[] beta) {
...
+    return append(Arrays.copyOf(ary,id), Arrays.copyOfRange(ary,id+1,ary.length));
+  }
+
+  public static long[] remove(long[] ary, int id) {
+    if(id < 0 || id >= ary.length) return Arrays.copyOf(ary,ary.length);
     if(id == ary.length-1) return Arrays.copyOf(ary,id);
-    if(id == 0) return Arrays.copyOfRange(ary,id,ary.length);
-    return append(Arrays.copyOf(ary,id), Arrays.copyOfRange(ary,id,ary.length));
+    if(id == 0) return Arrays.copyOfRange(ary,1,ary.length);
+    return append(Arrays.copyOf(ary,id), Arrays.copyOfRange(ary,id+1,ary.length));
```
Comment (arnocandel) [inline h2o-core/src/main/java/water/util/ArrayUtils.java]: thx

File: h2o-core/src/main/java/water/util/ArrayUtils.java [hunk 4]
```diff
@@ -1372,7 +1418,6 @@ public static int countNonzeros(double[] beta) {
         uniqueValidPoints[count++]= min;
         if (pos> min) uniqueValidPoints[count++]=pos;
         last=pos;
-        continue;
```
Comment (arnocandel) [inline h2o-core/src/main/java/water/util/ArrayUtils.java]: i assume this is now tested too

**Comment by vpatryshev** [comment]
Now it is. :)

Thanks,
-Vlad

**Pull request closed without merging**